common/XSIMDTypes.hpp

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2020 Nicholas Corgan
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <xsimd/types/xsimd_traits.hpp>
+
+#include <complex>
+#include <cstdint>
+#include <type_traits>
+
+//
+// This header is to be included by XSIMD implementation files. The
+// compile flags used for the file will determine which of these
+// macros will be defined. This allows us to use the SFINAE structs
+// below to only build SIMD implementations for types supported on the
+// given platform. If the type isn't supported, a fallback implementation
+// will be used.
+//
+// See: https://xsimd.readthedocs.io/en/latest/api/available_wrappers.html
+//
+
```

Comment:
Should PothosCore provided this? Looks like type detection that would be generally useful?

Reply:
I figured this was a detail of using XSIMD for the SIMD stuff, but with this being common (the same file is probably in PothosComms), I can see the argument for it. I'll move it over with some tweaks and add a "can CMake find this header" condition for building the relevant modules.